### PR TITLE
Biz portal user endpoint

### PR DESCRIPTION
**Pull request by esdd1995**
# Description

This PR includes the following proposed change(s):

[added endpoints for Delete and Get bizPortalUsers.]

Modified QueryAsync methods with switch in ProtalUserRepository.cs

abstraction used in Contract.cs to make PortalUserResp and PortalUserListResp inherit from PortalUserQryResp

NOTE: changes in BizProfileManager (castings) are because of QueryAsync modification.

### CODE REVIEW

File: src/Spd.Presentation.Licensing/Controllers/BizPortalUserController.cs [hunk 1]
```diff
@@ -85,4 +85,30 @@ public async Task<BizPortalUserListResponse> GetBizPortalUserList([FromRoute] Gu
...
         return await _mediator.Send(new BizPortalUserListQuery(bizId));
     }
+
+    /// <summary>
+    /// Get Business Portal User
+    /// </summary>
+    /// <param name="bizId"></param>
+    /// <returns></returns>
+    [Authorize(Policy = "OnlyBCeID", Roles = "PrimaryManager,Manager")]
+    //TODO: ask about the route address, should it be '/business' instead of 'bizs'?
```
Comment (carolcarpenter) [inline src/Spd.Presentation.Licensing/Controllers/BizPortalUserController.cs:95]: Yes

File: src/Spd.Manager.Licence/BizPortalUserManager.cs [hunk 2]
```diff
@@ -110,4 +120,37 @@ public async Task<BizPortalUserListResponse> Handle(BizPortalUserListQuery query
...
+               new PortalUserUpdateLoginCmd(cmd.UserId),
+               ct);
+        return default;
+    }
+
+    public async Task<Unit> Handle(BizPortalUserDeleteCommand request, CancellationToken cancellationToken)
+    {
+        //check role max number rule
+        
+        //TODO: ask if OrgId should = BizId
```
Comment (carolcarpenter) [inline src/Spd.Manager.Licence/BizPortalUserManager.cs:136]: Yes

File: src/Spd.Manager.Licence/BizPortalUserManager.cs [hunk 3]
```diff
@@ -110,4 +120,37 @@ public async Task<BizPortalUserListResponse> Handle(BizPortalUserListQuery query
...
+        var existingUsersResult = (PortalUserListResp) await _portalUserRepository.QueryAsync(
+            new PortalUserQry() { OrgId = request.BizId},
+            cancellationToken);
+        var toDeleteUser = existingUsersResult.Items.FirstOrDefault(u => u.Id == request.UserId);
+        var newUsers = existingUsersResult.Items.ToList();
+        if (toDeleteUser == null) return default;
+        newUsers.Remove(toDeleteUser);
+        var biz = (BizResult) await _bizRepository.QueryBizAsync(new BizsQry(request.BizId), cancellationToken);
+
+        //TODO: ask what is primaryUserNo and how to fill ContactRoleCode (or ContactAuthorizationTypeCode)
```
Comment (carolcarpenter) [inline src/Spd.Manager.Licence/BizPortalUserManager.cs:146]: There is a 'MaximumNumberOfAuthorizedContacts' and a 'MaximumNumberOfPrimaryAuthorizedContacts' that should be enforced for each business.

Comment (esdd1995) [reply src/Spd.Manager.Licence/BizPortalUserManager.cs:146]: where should exactly enforce it? there is a func at line 149 `CheckMaxRoleNumberRuleAsync`

Comment (carolcarpenter) [reply src/Spd.Manager.Licence/BizPortalUserManager.cs:146]: You could do it the same way it is done in Screening... check OrgUserManager.cs... it calls 'SharedManagerFuncs.CheckMaxRoleNumberRuleAsync'

File: src/Spd.Presentation.Licensing/Controllers/BizPortalUserController.cs (hunk 1 above)
Comment (peggy-quartech) [reply src/Spd.Presentation.Licensing/Controllers/BizPortalUserController.cs:95]: It seems everyone wants to change name from biz to business. For me, I think business is too long. But if you guys want, we can change all "biz" to business. Up to you guys.

Comment (esdd1995) [reply src/Spd.Presentation.Licensing/Controllers/BizPortalUserController.cs]: I just noticed every other routes are lik api/business. I'm Ok with either. @peggy-quartech

**Review by carolcarpenter — APPROVED** [review]

**Pull request merged**